src/Components/OATHeader/OATHeader.tsx

```diff
@@ -316,34 +313,55 @@ const OATHeader = ({ dispatch, state }: OATHeaderProps) => {
     };
 
     const handleFileListChanged = async (files: Array<File>) => {
-        const items = [];
+        const newModels = [];
         if (files.length > 0) {
             const filesErrors = [];
             let modelsMetadataReference = null;
             for (const current of files) {
                 const content = await current.text();
-                const error = await parseModel(content);
-                if (!error) {
+                const JSONContent = JSON.parse(content);
+                newModels.push(JSONContent);
+            }
+
+            const combinedModels = [...models, ...newModels];
+            const error = await parseModels(combinedModels);
+
+            if (!error) {
+                console.log('No error');
+                let loopIndex = 0;
+                for (const current of files) {
```

Comment:
It might be cleaning to use a traditional `for (let i = 0; i <...` rather than have the `loopIndex` as well.